### PR TITLE
fix(lovelace): key strategy resource cleanup off collection presence

### LINKED ISSUE #706: Strategy resource cleanup is dead after a restart: "resources" flag never set on already_registered path
Found while reviewing #702. Part of the umbrella issue #699 area, but **not** introduced by #702 — see "Pre-existing" below.

## Problem

`async_register_strategy_resource` sets `hass.data[DOMAIN]["resources"] = True` only on the `async_create_item` branch (`custom_components/keymaster/resources.py` line 67). The `already_registered` early return (lines 48-50) does not set it:

```python
    already_registered = any(data[CONF_URL] == STRATEGY_PATH for data in resources.async_items())

    if already_registered:
        _LOGGER.debug("Strategy module already registered")
        return
```

After a Home Assistant restart the strategy resource is already present in `.storage/lovelace_resources`, so registration short-circuits on that early return, and the flag stays `False` for the entire HA run.

`async_cleanup_strategy_resource` then bails at its guard (line 85):

```python
    if not hass_data.get("resources"):
        _LOGGER.debug("Strategy module not automatically registered, skipping removal")
        return
```

so `async_delete_item` is never awaited and the resource is never removed when the last keymaster entry unloads. In practice the cleanup path is effectively dead after the first restart: it can only ever fire within the same HA run in which the resource was originally created.

## Verification

Confirmed with an executable probe (a throwaway pytest, since discarded), not by reading alone. With the resource pre-present in the collection to simulate a post-restart state:

- `async_register_strategy_resource(hass)` leaves `hass.data[DOMAIN]["resources"]` as `False`.
- `async_cleanup_strategy_resource(hass, hass.data[DOMAIN])` returns without awaiting `async_delete_item`; the resource remains in the collection.

## Pre-existing

This is not a regression from #702. Lines 67 and 85 are byte-identical at merge base `83a572ea` (`git show 83a572ea:custom_components/keymaster/resources.py`). #702 only adds `hass_data["resources"] = False` after the delete, which is correct in itself but does not address the flag never being set on the `already_registered` path.

## Suggested direction

Either set the flag on the `already_registered` path as well, or drop the in-memory flag entirely and make cleanup idempotent by keying off the presence of `STRATEGY_PATH` in the resource collection (guarding against removing a resource a user added manually in YAML mode, which the existing `ResourceYAMLCollection` branch already handles).

### PR DESCRIPTION
**Pull request by tykeal**
# Summary

Strategy resource cleanup was effectively dead after the first Home Assistant
restart. `async_register_strategy_resource` only set the in-memory
`hass.data[DOMAIN]["resources"]` flag on the `async_create_item` branch; the
`already_registered` early return left it falsy. After a restart the resource
is already present in `.storage/lovelace_resources`, so registration
short-circuits and the flag never flips for the entire run. The cleanup guard
then bailed and `async_delete_item` was never awaited, so the strategy
resource lingered forever once the last keymaster entry unloaded.

This drops the unreliable in-memory flag entirely and makes cleanup idempotent
by keying off the actual presence of `STRATEGY_PATH` in the resource
collection.

## Proposed change

**Why remove the flag rather than patch it:** setting the flag on the
`already_registered` path too would fix one symptom, but the flag is
in-memory state that cannot survive a restart — it is unreliable by design.
Keying cleanup off the actual collection contents is correct by construction
and cannot be defeated by a restart.

- Remove all reads and writes of the `"resources"` flag:
  - the write in `async_register_strategy_resource` (`resources.py`),
  - the two guards and the reset write in `async_cleanup_strategy_resource`
    (`resources.py`),
  - the `{"resources": False}` seed in `async_setup` (`__init__.py`).
  - The unrelated `"resources_warned"` / `"yaml_warned"` log-once throttles
    are left untouched.
- **`hass_data` parameter decision:** removing the flag makes the
  `hass_data: dict[str, Any]` argument of `async_cleanup_strategy_resource`
  unused. It is dropped rather than kept as a dead argument. This is an
  internal helper with a single caller — `async_unload_entry` in
  `__init__.py` — which is updated accordingly. There is no external/public
  API to preserve, and leaving the unused parameter would be flagged by Ruff.
- **YAML-mode guard preserved:** cleanup still refuses to auto-remove in
  `ResourceYAMLCollection` mode. The "resources are in YAML mode, skipping
  automatic removal" debug log now fires when `STRATEGY_PATH` is actually
  present in the collection, preserving the equivalent signal without relying
  on the removed flag.

**Call sites of the two functions** (`async_register_strategy_resource`,
`async_cleanup_strategy_resource`): both live in `__init__.py`. Registration
is called in `async_setup` and `async_setup_entry` (unchanged signature).
Cleanup is called once, in `async_unload_entry`, updated to drop `hass_data`.

### Behavior change (please note)

In **storage** mode, cleanup now also removes a `STRATEGY_PATH` resource that
a user added manually through the Lovelace UI, because once the flag is gone
it is no longer distinguishable from one keymaster registered. This is
accepted: the strategy module is inert without keymaster installed, so
removing it on the last-entry unload is harmless. **YAML-mode** resources
remain user-managed and are never auto-removed.

### Tests

The throwaway probe from issue #706 is reinstated as permanent coverage in
`tests/test_resources.py`:

1. **Post-restart regression** (`..._removes_after_restart`): with
   `STRATEGY_PATH` already present, registration short-circuits
   (`async_create_item` not called) and cleanup **does** await
   `async_delete_item`. This test **fails on the pre-fix code** and passes
   after the fix.
2. Fresh registration → cleanup removes it (same-run path still works).
3. YAML mode with the resource present → cleanup does **not** remove, logs the
   YAML-mode skip.
4. YAML mode with the resource absent → silent, no crash.
5. No Lovelace / no resource collection → no crash.
6. `STRATEGY_PATH` absent from the collection → cleanup is a no-op.

Fails-before / passes-after was verified explicitly by running the regression
against the pre-fix source (resource **not** removed) and after (resource
removed). `resources.py` retains 100% line coverage; `tox -e lint`
(ruff, ruff format, mypy, codespell) is clean.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #706
- This PR is related to issue: #699

### CODE REVIEW

File: tests/test_resources.py [hunk 1]
```diff
@@ -185,56 +184,85 @@ async def test_register_strategy_resource_yaml_mode_subsequent_debug(
...
-    """Test cleaning up strategy resource."""
+    """Test cleaning up strategy resource removes it from the collection."""
     mock_storage_resources.async_items.return_value = [{"id": "resource_id", "url": STRATEGY_PATH}]
     hass.data[LOVELACE_DOMAIN] = MagicMock()
     hass.data[LOVELACE_DOMAIN].resources = mock_storage_resources
 
-    await async_cleanup_strategy_resource(hass, {"resources": True})
+    await async_cleanup_strategy_resource(hass)
 
     mock_storage_resources.async_delete_item.assert_called_once_with("resource_id")
```
[the remark this answers is not in the document]
Comment (tykeal) [reply tests/test_resources.py]: Good catch. Switched both positive delete assertions (lines 194 and 220) to `assert_awaited_once_with("resource_id")` so the tests verify the coroutine is actually awaited, which is exactly the regression under test. Pushed in bda8429.

**Review by secondof9 — APPROVED** [review]
## 📋 Review Summary
> [!TIP]
> **Review Status:** 🟢 APPROVED
> **Change Type:** 🐛 Bug Fix
> **Review Effort:** 🟢 Low
> **Core Impact:** Removes an unreliable in-memory `resources` flag from Keymaster's Lovelace strategy cleanup and keys cleanup off the actual presence of `STRATEGY_PATH` in the resource collection, fixing issue #706 (resource lingering forever after an HA restart). Prior Copilot feedback is fully addressed and all CI checks pass.

---

## 🚦 CI & Pipeline Health Summary

| Check / Workflow Name | Status | Impact on Review |
| :--- | :--- | :--- |
| Pytest (3.14) | ✅ PASSED | Full test suite green on Python 3.14 |
| coverage | ✅ PASSED | All modified lines covered; project 94.97% |
| Hassfest Validation | ✅ PASSED | Integration manifest compliant |
| HACS Validation | ✅ PASSED | Custom component validation clean |
| Prek | ✅ PASSED | Pre-commit linters clean |
| Autolabel PR (×2) | ✅ PASSED | Labels applied |

> [!NOTE]
> **CI Pipeline Clear:** All 7 check runs reached `completed` with `success`/`skipped` conclusions. No pending or failing workflows.

---

## 🔍 Architectural Walkthrough

### `custom_components/keymaster/resources.py` — core fix

* **Root cause correctly diagnosed:** `async_register_strategy_resource` only set `hass.data[DOMAIN]["resources"] = True` on the `async_create_item` branch. After a restart, `.storage/lovelace_resources` already contains `STRATEGY_PATH`, so the `already_registered` early return (line 50) short-circuits and the flag is never set for the entire run. The old cleanup guard then bailed and `async_delete_item` was never awaited, so the resource lingered forever on the last-entry unload.
* **The fix is correct by construction:** the unreliable in-memory flag is dropped entirely. Cleanup now keys off `any(data[CONF_URL] == STRATEGY_PATH for data in resources.async_items())` — a deterministic read of actual collection state that cannot be defeated by a restart. `resources_warned`/`yaml_warned` log-once throttles are preserved via `.get()` defaults (lines 34, 55), so no KeyError risk.
* **`hass_data` parameter removal is justified:** removing the flag makes `hass_data: dict[str, Any]` unused. It's a single-caller internal helper (`async_unload_entry`, line 353), updated to drop the argument. Leaving it would be a genuine Ruff `ARG001`/unused-arg violation, so dropping it is the right call rather than keeping dead state.
* **YAML-mode guard preserved:** the `ResourceYAMLCollection` branch still refuses to auto-remove, now logging the skip when `STRATEGY_PATH` is actually present. User-managed YAML resources remain untouched.
* **Async boundary is clean:** `resources.async_items()` is iterated consistently with the existing pattern at line 48; `async_delete_item` and `async_load` remain awaited. No blocking I/O, no sync `requests`.

### `custom_components/keymaster/__init__.py` — call-site updates

* **`async_setup` seed** (line 95): `hass.data.setdefault(DOMAIN, {})` replaces `"resources": False`. No dangling reads of the removed key remain anywhere in the file.
* **`async_unload_entry`** (line 343-353): cleanup still runs only when no other keymaster entries remain (`remaining_entries` check), then calls the new single-arg signature. Correct lifecycle placement.
* **`delete_coordinator`** (line 391): retains its own independent `hass.data.get(DOMAIN, {})` read for coordinator teardown — unaffected by the removed flag. No accidental coupling.

### `tests/test_resources.py` — regression coverage

* **`test_cleanup_strategy_resource_removes_after_restart`** is the key addition and is a genuine fails-before/passes-after test: it seeds `STRATEGY_PATH` present, calls registration (which short-circuits — `async_create_item` never called), then asserts cleanup **does** await `async_delete_item`. This is exactly the regression under test and it now uses `assert_awaited_once_with`.
* **`assert_awaited_once_with` on lines 194 and 220** directly resolves the prior Copilot `🟡 Changes recommended` feedback (thread resolved in commit `bda8429`). This proves the coroutine is truly awaited, not merely called.
* **Edge cases covered:** YAML present (skips + logs), YAML absent (silent), no resources (no-op), no Lovelace collection (no crash). `hass.data[DOMAIN] = {}` is set in each setup test, consistent with the new seed.

---

## ✅ Prior Feedback Resolution

The initial Copilot review (`🟡 Changes recommended`, review #5083903155) flagged that `assert_called_once_with` for the `AsyncMock` deletes doesn't prove `async_delete_item` was awaited — the very regression this PR prevents. The author addressed this in commit `bda8429` by switching both positive delete assertions (lines 194, 220) to `assert_awaited_once_with`, and the thread is now resolved. No outstanding review items remain.

## 🔗 Related / Outside-Diff Context

* **Behavior change (documented in PR body):** in storage mode, cleanup now also removes a `STRATEGY_PATH` resource a user added manually via the Lovelace UI, since it's no longer distinguishable from a keymaster-registered one. This is accepted and correct — the strategy module is inert without keymaster installed, so removal on last-entry unload is harmless. YAML-mode resources remain user-managed and are never auto-removed.
* **Coverage note:** Codecov reports 94.97% (down slightly from a fully-covered baseline elsewhere, but all modified and coverable lines here are covered). `resources.py` retains 100% line coverage per the author.

**Review by firstof9 — APPROVED** [review]
Looks good.
